docs/proposals/20230123-azure-service-operator.md

```diff
@@ -0,0 +1,214 @@
+
+During the initial adoption of ASO, CAPZ's ASO-based [reconciliation flow](#reconciliation) will automatically adopt resources previously managed by CAPZ based on the `sigs.k8s.io_cluster-api-provider-azure_cluster_*` tag on the Azure resource.
+
+After that, a CAPZ upgrade may include a bump in the API version used for some ASO resource types. ASO will handle those changes transparently from CAPZ's perspective with its own conversion webhooks. The limit on the size of Kubernetes resources will likely force ASO to remove certain API versions of resources in CRDs in the future. When a version is deprecated in ASO, CAPZ will have to upgrade its use of the resource to a newer API version.
+
+Separate from API version upgrades of resources, upgrades to ASO itself may be included in new versions of CAPZ. [ASO's documentation](https://azure.github.io/azure-service-operator/introduction/upgrading/) recommends upgrading using the same mechanism used to initially install. At the same time when CAPZ installs ASO, it will also upgrade an existing ASO it installed if necessary.
+
+## Additional Details
+
+### Test Plan
+
+Adoption of ASO to manage Azure infrastructure is at the core of CAPZ's functionality. As such, changes made to integrate ASO must be thoroughly tested.
+
+e2e tests will give the best indication if any regressions are introduced. Because no user-facing changes to CAPZ are expected as a result of integrating with ASO, all of the existing e2e tests will not need any changes to continue providing the same value they do already and do not need to compensate for the transition period where ASO is only partially integrated. New tests or additions to the test framework may include the following:
+- a checkpoint when installing CAPZ in the framework that ASO is also installed and healthy
+- upgrade tests from a version of CAPZ that does not use ASO to one that does,
+  verifying that CAPZ-managed and unmanaged Azure resources become correctly
+  reflected in ASO after the upgrade
+- a test to verify when a CAPZ upgrade includes an upgrade of the ASO control plane
+- a test to verify when a CAPZ upgrade includes a newer API version of an ASO resource
+
+To catch performance and scalability regressions, one-off tests against CAPZ instances managing many and/or large workload clusters will also be performed. Telemetry will be inspected to ensure the time it takes to reconcile a high volume of clusters and the overhead required in the management cluster to run ASO are still acceptable.
+
+Finally, unit tests will be added to cover tightly-scoped functional changes related to ASO adoption, like error handling.
+
+### Graduation Criteria
+
+ASO integration will not be kept behind a feature flag or matriculate through the usual alpha, beta, and stable phases. Instead, the transition will be made one Azure service interface at a time so as to distribute potential impact over time.
```

Comment:
this section could also detail the rollout/rollback plan service-per-service (even if the order or timeline itself is TBD)